pl_bolts/datasets/array_dataset.py

```diff
@@ -0,0 +1,68 @@
+from typing import Callable, List, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from pytorch_lightning.utilities import apply_func, exceptions
+
+from pl_bolts.datasets.base_dataset import BoltsDataset
+
+# TODO: Move this to a more appropriate place.
+ARRAYS = Union[torch.Tensor, np.ndarray, List[Union[float, int]]]
+
+
+class ArrayDataset(BoltsDataset):
+    """Dataset wrapping tensors, lists, numpy arrays.
+
+    Args:
+        arrays: Sequence of indexables.
+        transforms: A callable that takes input data and target data and returns transformed versions of both.
```

Comment:
Currently, a Misconfiguration exception is raised in the base class when transforms or input/target transforms are both not None. I took the idea from the VisionDataset. I agree that it's confusing and that it would be easier to remove the transforms arg 👍🏼